purge instructions

jchancellor-ms opened the pull request:
## Purpose
Updates the purge instruction.  Also modifies the deploy instructions so that the items that require text edits can be copied as a single line.
* ...

## Does this introduce a breaking change?
<!-- Mark one with an "x". -->
```
[ ] Yes
[ x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code
<!-- Add steps to run the tests suite and/or manually test -->
```
```

## What to Check
Verify that the following are valid
* ...

## Other Information
<!-- Add any other helpful information that may be needed here. -->

jchancellor-ms committed:
purge instructions

ckittel commented on README.md:
```diff
@@ -66,13 +66,19 @@ The following steps are required to deploy the infrastructure from the command l
 
-1. Create a resource group and deploy the infrastructure.
-
+1. Set the deployment location. Because this solution uses Azure AI Studio, the location MUST be one of: `southcentralus`, `westeurope`, `southeastasia`, `japaneast` to support all resources deployed
```
```suggestion
1. Set the deployment location.

   Because this solution uses Azure AI Studio, the location MUST be one of: `southcentralus`, `westeurope`, `southeastasia`, `japaneast` to support all resources deployed.

```

ckittel commented on README.md:
````diff
@@ -66,13 +66,19 @@ The following steps are required to deploy the infrastructure from the command l
+   ```
+
+1. Set the base name value that will be used as part of the Azure resource names for the resources deployed in this solution.
+   ```bash
````
~~~suggestion

   ```bash
~~~

ckittel commented on README.md:
```diff
@@ -201,15 +207,15 @@ After the deployment is complete, you can try the deployed application by naviga
 
 Most of the Azure resources deployed in the prior steps will incur ongoing charges unless removed. Also a few of the resources deployed go into a soft delete status. It's best to purge those once you're done exploring, Key Vault is given as an example here. Azure OpenAI and Azure Machine Learning Workspaces are others that should be purged.
 
+> **_NOTE:_** To purge the AI services resources, `Microsoft.CognitiveServices/locations/resourceGroups/deletedAccounts/delete` permission must be given to the user on the subscription scope where the resources are deployed. A role assignment at the resource or resource group scopes will be insufficient to access the purge functionality. See the [learn documentation](https://learn.microsoft.com/e ...
```
```suggestion
> **_NOTE:_** To purge the AI services resources, `Microsoft.CognitiveServices/locations/resourceGroups/deletedAccounts/delete` permission must be given to the user on the subscription scope where the resources are deployed. A role assignment at the resource or resource group scopes will be insufficient to access the purge functionality. See the [learn documentation](https://learn.microsoft.com/azure/ai-services/recover-purge-resources?tabs=azure-cli#purge-a-deleted-resource) for additional details.
```

ckittel commented on README.md:
```diff
@@ -201,15 +207,15 @@ After the deployment is complete, you can try the deployed application by naviga
 
 Most of the Azure resources deployed in the prior steps will incur ongoing charges unless removed. Also a few of the resources deployed go into a soft delete status. It's best to purge those once you're done exploring, Key Vault is given as an example here. Azure OpenAI and Azure Machine Learning Workspaces are others that should be purged.
 
+> **_NOTE:_** To purge the AI services resources, `Microsoft.CognitiveServices/locations/resourceGroups/deletedAccounts/delete` permission must be given to the user on the subscription scope where the resources are deployed. A role assignment at the resource or resource group scopes will be insufficient to access the purge functionality. See the [learn documentation](https://learn.microsoft.com/e ...
```
How is this not already covered by the permissions listed in the pre-preq?

jchancellor-ms replied:
We would just need to be more explicit on the scope.  Since we have them create a resource group that can probably be inferred.  I'll add text to explicitly state that the permissions should be at the subscription level.

ckittel replied:
Yeah, if they cannot create a resource group, they couldn't proceed (it's very intentional that we make that a step, vs saying "choose an existing empty resource group") -- helps us better scope permissions.  If you can still include some bits about the scope in the pre-reqs, that would be great!

ckittel commented on README.md:
```diff
@@ -201,15 +207,15 @@ After the deployment is complete, you can try the deployed application by naviga
-az ai services purge -n aih-${BASE_NAME} <--- TODO
-Role assignment removals
+SUBSCRIPTION_ID=$(az account show --query id --output tsv)
+az keyvault purge -n kv-${BASE_NAME} -l $LOCATION --no-wait
```
Why is location needed now for key vault?  Was it always needed and the old command had a typo?

jchancellor-ms replied:
The command reference says both name and location are optional which I think is a documentation error.  (Name should be required at a minimum.)  I included both for completeness.  In practice name alone would probably be sufficient.

ckittel replied:
yeah, key vault names are globally unique.  we use the previous command in lots of places without any issue.  I'd suggest just reverting as one less variable for someone to get wrong in this command :)

ckittel commented on README.md:
```diff
@@ -201,15 +207,15 @@ After the deployment is complete, you can try the deployed application by naviga
-Role assignment removals
+SUBSCRIPTION_ID=$(az account show --query id --output tsv)
+az keyvault purge -n kv-${BASE_NAME} -l $LOCATION --no-wait
+az resource delete --ids /subscriptions/${SUBSCRIPTION_ID}/providers/Microsoft.CognitiveServices/locations/${LOCATION}/resourceGroups/${RESOURCE_GROUP}/deletedAccounts/aih-${BASE_NAME}
```
```suggestion
az resource delete --ids $(az cognitiveservices account show -n oai-${BASE_NAME} -g $RESOURCE_GROUP --query 'id' -o tsv)
```

And then remove `SUBSCRIPTION_ID`

ckittel replied:
Maybe more importantly, why not just `az cognitiveservices account purge` ?

jchancellor-ms replied:
So the recommended change may not work, because in theory the resource group delete would have already removed the resource so the az cognitiveservices account show shouldn't find it.  I'll try the purge option even though the official documentation suggests to use the resource delete option.  If it works I'll update to use the account purge as it's easier to read.

ckittel committed:
Apply suggestions from code review

ckittel committed:
Merge branch 'collab/ai-studio' into purge

ckittel commented on README.md:
```diff
@@ -210,15 +220,15 @@ Once you're there, ask your solution a question. Like before, you question shoul
 
 Most of the Azure resources deployed in the prior steps will incur ongoing charges unless removed. Also a few of the resources deployed go into a soft delete status. It's best to purge those once you're done exploring, Key Vault is given as an example here. Azure OpenAI and Azure Machine Learning Workspaces are others that should be purged.
 
+> **_NOTE:_** To purge the AI services resources, `Microsoft.CognitiveServices/locations/resourceGroups/deletedAccounts/delete` permission must be given to the user on the subscription scope where the resources are deployed. A role assignment at the resource or resource group scopes will be insufficient to access the purge functionality. See the [learn documentation](https://learn.microsoft.com/a ...
+
```
I don't think this note is relevant in this scenario.  We create resource directly as part of this, and the prereqs already cover this operation.  Am I missing something?

```suggestion
```

ckittel commented on README.md:
```diff
@@ -210,15 +220,15 @@ Once you're there, ask your solution a question. Like before, you question shoul
 
 Most of the Azure resources deployed in the prior steps will incur ongoing charges unless removed. Also a few of the resources deployed go into a soft delete status. It's best to purge those once you're done exploring, Key Vault is given as an example here. Azure OpenAI and Azure Machine Learning Workspaces are others that should be purged.
```
You should update this line to reflect the new state of the situation.  Start with the following and tweak as you see fit.

```suggestion
Most of the Azure resources deployed in the prior steps will incur ongoing charges unless removed. Also a few of the resources deployed go into a soft delete status. It's best to purge those once you're done exploring.
```

ckittel reviewed (changes requested):
Sorry for the multiple "reviews" -- GitHub and I are not friends today :P

jchancellor-ms committed:
Updated purge instructions

Used the az commands, updated the pre-req language to remove the additional purge text.

ckittel commented on README.md:
```diff
@@ -42,7 +42,9 @@ Follow these instructions to deploy this example to your Azure subscription, try
   - `Microsoft.Storage`
 
-- Your user has permissions to assign [Azure roles](https://learn.microsoft.com/azure/role-based-access-control/built-in-roles), such as a User Access Administrator or Owner.
+- Your deployment user has permissions to assign [Azure roles](https://learn.microsoft.com/azure/role-based-access-control/built-in-roles) on newly created resource groups and resources. (E.g. `User Access Administrator` or `Owner`) 
+
+- Your deployment user has permissions to purge AI services resources at the subscription scope where the AI services resources are deployed. (E.g. `Contributor` or `Cognitive Services Contributor` built-in roles, or a custom role with the `Microsoft.CognitiveServices/locations/resourceGroups/deletedAccounts/delete` permission)
```
```suggestion
- Your deployment user must have the following permissions at the subscription scope:

  - Ability to assign [Azure roles](https://learn.microsoft.com/azure/role-based-access-control/built-in-roles) on newly created resource groups and resources. (E.g. `User Access Administrator` or `Owner`) 
  - Ability to purge deleted AI services resources. (E.g. `Contributor` or `Cognitive Services Contributor`)
```

ckittel commented on README.md:
```diff
@@ -215,15 +222,12 @@ Once you're there, ask your solution a question. Like before, you question shoul
-az keyvault purge -n kv-${BASE_NAME} -l $LOCATION --no-wait
-az resource delete --ids /subscriptions/${SUBSCRIPTION_ID}/providers/Microsoft.CognitiveServices/locations/${LOCATION}/resourceGroups/${RESOURCE_GROUP}/deletedAccounts/aih-${BASE_NAME}
+az keyvault purge -n kv-${BASE_NAME} -l $LOCATION 
+az cognitiveservices account purge --location $LOCATION --name oai-${BASE_NAME} --resource-group $RESOURCE_GROUP
```
```suggestion
az cognitiveservices account purge -g $RESOURCE_GROUP -l $LOCATION --name oai-${BASE_NAME}
```

ckittel committed:
Apply suggestions from code review

jchancellor-ms committed:
purge text changes

Updated the purge instruction descriptions

jchancellor-ms committed:
Merge branch 'purge' of https://github.com/jchancellor-ms/openai-end-to-end-basic into pr/17
